ipfs-pinning-service.yaml

```diff
@@ -1,40 +1,69 @@
 openapi: 3.0.0
 info:
-  version: "0.0.1"
-  title: 'IPFS Pinning Service API'
-  description: "This is the IPFS Pinning Service API. It attempts to be an
+  version: "0.0.2"
+  title: '(WIP) IPFS Pinning Service API'
+  description: "# About
+
+    This is the IPFS Pinning Service API. It attempts to be an
     implementation-agnostic API designed to be implemented by Pinning Service
-    Providers.
+    Providers and used in client mode by IPFS nodes and GUI applications.
+
+    ## THIS SPEC IS WORK IN PROGRESS 🏗️
+
+    Join design discussion at  [https://github.com/ipfs/pinning-services-api-spec](https://github.com/ipfs/pinning-services-api-spec)
+
+    # Pin object lifecycle
+
+    ## Creating pin
+
+    User creates pin object via `POST /pins` and receives `PinStatus` response
+
+    - `id` in `PinStatus` is `cid-of-pin-object` that can be used for modifying and removing pin in the future
+
+    - `status` in  `PinStatus` indicates current state of a pin
+
+    ### Checking status of pinning in progress
+
+    `status` from `PinStatus` may indicate a pending state. It means the data behind `Pin.cid` was not found on pinning service and is being fetched from IPFS network, which may take time.
+
+    In that case, user can periodically check pinning progress via `GET /pins/{cid-of-pin-object}`, until pinning is successful, or decides to remove pending pin.
 
+    ## Modifying pin object
 
-    This specification does **NOT** define the IPFS pinning API.
+    User may decide to modify existing pin object via `POST /pins/{cid-of-pin-object}`. The new pin object `id` is returned in `PinStatus` response. The old pin object is deleted automatically.
 
+    ## Removing pin object
 
-    GitHub discussion around this API can be found in [ipfs/pinning-services-api-spec](https://github.com/ipfs/pinning-services-api-spec).
+    Pin object can be removed via `DELETE /pins/{cid-of-pin-object}`
 
+    # Authorization
 
+    An opaque auth token is required to be sent with each request.
+    There are two ways of doing so:
 
-    **Critical Notice**
+    - HTTP header: `Authorization: Bearer <auth>`
 
+    - query parameter: `&auth=<auth>`
 
-    To allow backend systems behind the API to associate pins with a particular
-    user, the API requires that a JSON Web Token is passed with requests that
-    contains a `sub` claim (as specified by
-    [OpenIDConnect Standard Claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims))
-    that represents the user's identity within the backend system, so as to
-    associate the user with their pins. This allows the backend system to use
-    the identity as a form of reference counting on pins, therefore only
-    removing pins once no other users have it pinned."
+  "
 
 servers:
   - url: https://api.example.com
+
 paths:
   /pins:
     get:
-      summary: Get all pins
+      summary: List pin objects
```

Comment:
Since we can pass `skip` and `limit` here, is there any way to know how many items there are in total?

Reply:
I'm fairly sure we'd be able to return a "total count" of items that matched a query, but I'm not sure about other services.